core/com.b2international.snowowl.datastore.server.tests/src/com/b2international/snowowl/datastore/server/internal/branch/BranchTest.java

```diff
@@ -136,6 +136,44 @@ public void testBehindState() throws Exception {
 	public void testDivergedState() throws Exception {
 		assertState(commit(branchA), commit(main), BranchState.DIVERGED);
 	}
+	
+	@Test
+	public void testValidBranchName() {
+		// no exception should be thrown
+		final String tempBranchName = createTempBranchName("TEST");
+		Branch.BranchNameValidator.DEFAULT.checkName(tempBranchName);
+	}
+	
+	@Test(expected = BadRequestException.class)
+	public void testInvalidBranchName() {
+		final String tempBranchName = createTempBranchName("TEST$$");
+		Branch.BranchNameValidator.DEFAULT.checkName(tempBranchName);
+	}
+	
+	@Test
+	public void testNameExtractionFromTemporaryBranch() {
```

Comment:
Please add test methods for the following scenarios (if there are no such methods):
Branch.BranchNameValidator.DEFAULT.getName():
- test success with branch name
- test success with temporary branch name
- test failure with invalid name, either empty or malformed
- test failure with invalid temporary branch name like '$TEST'